lxd/db_images.go

```diff
@@ -105,8 +105,17 @@ func dbImageSourceGet(db *sql.DB, imageId int) (int, api.ImageSource, error) {
 
 }
 
-// dbImageGet gets an ImageBaseInfo object from the database.
-// The argument fingerprint will be queried with a LIKE query, means you can
+// Whether an image with the given fingerprint exists.
+func dbImageExists(db *sql.DB, fingerprint string) (exists bool, err error) {
```

Comment:
Indeed, that's why the new function also returns err beside the boolean value.
Also, in your code, if you SELECT COUNT() you always get exactly one row, so the ErrNoRows is never triggered.